core/domain/skill_services.py

```diff
@@ -514,3 +515,82 @@ def save_skill_summary(skill_summary):
     )
 
     skill_summary_model.put()
+
+
+def create_user_skill_mastery(user_id, skill_id, degree_of_mastery):
+    """Creates skill mastery of a user.
+
+    Args:
+        user_id: str. The user ID of the user for whom to create the model.
+        skill_id: str. The unique id of the skill.
+        degree_of_mastery: float. The degree of mastery of user in the skill.
+    """
+
+    user_skill_mastery = skill_domain.UserSkillMastery(
+        user_id, skill_id, degree_of_mastery)
+    save_user_skill_mastery(user_skill_mastery)
+
+
+def save_user_skill_mastery(user_skill_mastery):
+    """Stores skill mastery of a user.
+
+    Args:
+        user_skill_mastery: dict. The user skill mastery model of a user.
+    """
+    user_skill_mastery_model = user_models.UserSkillMasteryModel(
+        id=user_models.UserSkillMasteryModel.construct_model_id(
+            user_skill_mastery.user_id, user_skill_mastery.skill_id),
+        user_id=user_skill_mastery.user_id,
+        skill_id=user_skill_mastery.skill_id,
+        degree_of_mastery=user_skill_mastery.degree_of_mastery)
+
+    user_skill_mastery_model.put()
+
+
+def get_skill_mastery(user_id, skill_id):
+    """Fetches the mastery of user in a particular skill.
+
+    Args:
+        user_id: str. The user ID of the user.
+        skill_id: str. Unique id of the skill for which mastery degree is
+            requested.
+
+    Returns:
+        degree_of_mastery: float. Mastery degree of the user for the
+            requested skill.
+    """
+    model_id = user_models.UserSkillMasteryModel.construct_model_id(
+        user_id, skill_id)
+    degree_of_mastery = user_models.UserSkillMasteryModel.get(
+        model_id)[0].degree_of_mastery
```

Comment:
Why [0]?

Reply:
because the returned object is a list with 1 element.

Reply:
See my comment for get() in gae_models.py. Typically get() is expected to retrieve a single element, not a list.

Reply:
yes I've fixed it. Thanks!